### PR TITLE
range index shouldn’t be loaded when the column is sorted

### PR DESCRIPTION
**Pull request by klsince**
This PR fixes a bug where an attempt is made to load a range index for sorted columns.

### CODE REVIEW

File: pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java [hunk 1]
```diff
@@ -123,13 +123,6 @@ public PhysicalColumnIndexContainer(SegmentDirectory.Reader segmentReader, Colum
       _bloomFilter = null;
     }
 
-    if (loadRangeIndex) {
```
Comment (richardstartin) [inline pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/column/PhysicalColumnIndexContainer.java:126]: Can’t you just add a !isSorted check here?

**Comment by richardstartin** [comment]
> Suggest changing line 80 to `boolean loadRangeIndex = !metadata.isSorted() && indexLoadingConfig.getRangeIndexColumns().contains(columnName);`

loadRangeIndex already had the second check baked into it. The only thing which needs addressing here is whether the column is sorted, so this should be a one line change.

**Comment by richardstartin** [comment]
I’d suggest changing the PR title and description - the range index _is_ loaded conditionally, it just shouldn’t be loaded when the column is sorted. _Before_ the range index just wasn’t loaded when there wasn’t a dictionary, regardless of configuration. This PR fixes a bug where an attempt is made to load a range index for sorted columns.

**Review by richardstartin — APPROVED** [review]

**Review by xiangfu0 — APPROVED** [review]
Thanks for fixing this!

**Pull request merged**